### ISSUE TITLE
getAnnotationCounts Acquisition

### ISSUE THREAD

**Issue by will-moore** [ome/omero-web#299]
See https://www.openmicroscopy.org/qa2/qa/feedback/30270/

If you select multiple Runs (Acquisitions) in the webclient tree, the right panel loading fails with

```conn.getAnnotationCounts() ```

```
File "/opt/omero/web/venv3/lib64/python3.6/site-packages/omeroweb/webclient/views.py", line 2288, in batch_annotate
getObjects(request, conn)

File "/opt/omero/web/venv3/lib64/python3.6/site-packages/omeroweb/webclient/controller/container.py", line 243, in getBatchAnnotationCounts
return self.conn.getAnnotationCounts(objDict)

File "/opt/omero/web/venv3/lib64/python3.6/site-packages/omero/gateway/__init__.py", line 3728, in getAnnotationCounts
return self.countAnnotations(obj_type, obj_ids)

File "/opt/omero/web/venv3/lib64/python3.6/site-packages/omero/gateway/__init__.py", line 3606, in countAnnotations
queryResult = self.getQueryService().projection(q, params, ctx)

omero.QueryException: exception ::omero::QueryException
{
serverStackTrace = ome.services.query.QueryException: Illegal query:
select sum( case when an.ns = :ratingns
and an.class = LongAnnotation
then 1 else 0 end),
sum( case when (an.ns is null or an.ns != :ratingns)
and an.class = LongAnnotation
then 1 else 0 end),
sum( case when an.class != LongAnnotation
then 1 else 0 end ), type(an.class)
from Annotation an where an.id in
(select distinct(ann.id) from AcquisitionAnnotationLink ial
join ial.child as ann
join ial.parent as i
where i.id in (:ids))
group by an.class

AcquisitionAnnotationLink is not mapped [
```

**Comment by Tom-TBT** [comment]
I just saw the error on my server. Never had more than one Run before until now (microscope data from a lab next to us, though wasn't supposed to be. Something wrong with the ome.xml).

So tracing back what happens:
`omeroweb.webclient.views.batch_annotate` calls getObjects(), returning a dictionary.
The key for the PlateAcquisition is "acquisition" https://github.com/ome/omero-web/blob/604d5279c96909a6862e4b1fc67de4f0a8699298/omeroweb/webclient/views.py#L2167

The dictionary is passed until reaching `omero.gateway.countAnnotations` where it fails.

Some checks are happening in  `omero.gateway.getAnnotationCounts`, replacing the key "plateacquisition" with "PlateAcquisition". That's where we could add one more condition to replace "acquisition" with "PlateAcquisition". Then the HQL shouldn't fail:
https://github.com/ome/omero-py/blob/0958d2ba30f92990d6dea077fa359356577c6275/src/omero/gateway/__init__.py#L3724

Another solution would be to fix the key in the dictionary passed to `getBatchAnnotationCounts`, but maybe the error happens in other places (so I prefer the first option)

**Comment by will-moore** [comment]
With https://github.com/ome/omero-web/pull/518 merged, this can be closed.

NB @Tom-TBT If you'd used the exact words "Fixes #299" on the PR, then merging it would have closed this issue (instead of "Fix of #299".
